src/blueprint/from_config/operators/http.rs

```diff
@@ -19,19 +19,43 @@ struct MustachePartsValidator<'a> {
   field: &'a FieldDefinition,
 }
 
-fn get_value_type(type_of: &config::Type, value: &str) -> Option<Type> {
-  if let Some(field) = type_of.fields.get(value) {
-    return Some(to_type(field, None));
-  }
-  None
-}
-
 impl<'a> MustachePartsValidator<'a> {
   fn new(type_of: &'a config::Type, config: &'a Config, field: &'a FieldDefinition) -> Self {
     Self { type_of, config, field }
   }
+  fn get_nested_type(&self, tail: &[String], is_query: bool) -> Result<(Type, String), String> {
+    let mut len = tail.len();
+    let mut type_of = self.type_of;
+    for item in tail {
+      if let Some(field) = type_of.fields.get(item) {
+        let val_type = to_type(field, None);
+
+        if len == 1 {
+          return Ok((val_type, item.to_owned()));
+        } else if !is_query && val_type.is_nullable() {
+          return Err(format!("value '{}' is a nullable type", item.as_str()));
+        }
+
+        type_of = self
+          .config
+          .find_type(&field.type_of)
+          .ok_or_else(|| format!("no type '{}' found", tail.join(".").as_str()))?;
+
+        len -= 1;
+      } else {
+        // get tail so far to be logged
+        let item = tail[0..tail.len() - len + 1].join(".");
+
+        return Err(format!("no value '{}' found", item.as_str()));
+      }
+    }
+
+    let item = tail[0..tail.len() - len + 1].join(".");
+
+    Err(format!("no value '{}' found", item.as_str()))
```

Comment:
as we can see on the codecov report, this code is unreachable (we have a lot of tests without value, but all of them are being handled on the loop above), but as recommended, I'm keeping the error return, just in case

Reply:
this did not work as the variable would start uninitialized :confused:

do you have any other suggestions?

Reply:
I've found a solution, I'm resolving this one